src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py

```diff
@@ -0,0 +1,718 @@
+        self._lang_url = "https://github.com/graalvm/setup-graalvm"
+        lang_distribution_exp = None
+        lang_version_exp = None
+        if distribution := get_step_input(step, key="distribution"):
+            if not is_expression(distribution):
+                self._lang_distributions = [distribution]
+            else:
+                lang_distribution_exp = distribution
+        if java_version := get_step_input(step, key="java-version"):
+            if not is_expression(java_version):
+                self._lang_versions = [java_version]
+            else:
+                lang_version_exp = java_version
+        # Handle matrix values.
+        matrix_values = {}
+        if lang_distribution_exp and "matrix." in lang_distribution_exp:
+            matrix_values["lang_distribution_var"] = find_expression_variables(
+                value=lang_distribution_exp, exp_var="matrix"
+            )
+        if lang_version_exp and "matrix." in lang_version_exp:
+            matrix_values["lang_version_var"] = find_expression_variables(value=lang_version_exp, exp_var="matrix")
+
+        if matrix_values:
+            job_node = external_node.caller
+            if job_node is None:
+                logger.debug("Unable to find the caller GitHub Action job for step %s.", external_node.name)
+                return
+            try:
+                if (variables := matrix_values.get("lang_distribution_var")) is not None:
+                    values: list[str] = []
+                    for var in variables:
+                        values.extend(resolve_matrix_variable(job_node, var))
+                    if values:
+                        self._lang_distributions = values
+            except GitHubActionsValueError as error:
+                logger.debug(error)
+
+            try:
+                if (variables := matrix_values.get("lang_version_var")) is not None:
+                    values = []
+                    for var in variables:
+                        values.extend(resolve_matrix_variable(job_node, var))
+                    if values:
+                        self._lang_versions = values
+            except GitHubActionsValueError as error:
+                logger.debug(error)
+
+    @property
+    def lang_name(self) -> str:
+        """Get the name of the language."""
+        return self._lang_name
+
+    @property
+    def lang_versions(self) -> list[str] | None:
+        """Get the possible version of the language."""
+        return self._lang_versions
+
+    @property
+    def lang_distributions(self) -> list[str] | None:
+        """Get the possible distributions of the language."""
+        return self._lang_distributions
+
+    @property
+    def lang_url(self) -> str | None:
+        """Get the URL that provides information about the language distributions and versions."""
+        return self._lang_url
+
+
+def create_third_party_action_model(external_node: GitHubWorkflowNode) -> ThirdPartyAction:
```

Comment:
This function is used at two locations:
- https://github.com/oracle/macaron/blob/ed96daea0f0c4ae81f6f42b464d859f0829563aa/src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py#L300
- https://github.com/oracle/macaron/blob/ed96daea0f0c4ae81f6f42b464d859f0829563aa/src/macaron/slsa_analyzer/ci_service/github_actions/analyzer.py#L251

One was for an external workflow, one was for a reuseable workflow. I wonder if we should rename this parameter be more generic? Or are we considering a `GHWorkflowType.REUSABLE` type workflow an "external" workflow too.

Reply:
An external workflow can be either a normal GitHub Action or a Reusable GitHub Action.